x/rollapp/keeper/msg_server_create_rollapp.go

```diff
@@ -26,9 +27,15 @@ func (k msgServer) CreateRollapp(goCtx context.Context, msg *types.MsgCreateRoll
 
 	if rollappId.IsEIP155() {
 		// check to see if the RollappId has been registered before with same key
-		if _, isFound := k.GetRollappByEIP155(ctx, rollappId.GetEIP155ID()); isFound {
+		rollapp, isFound := k.GetRollappByEIP155(ctx, rollappId.GetEIP155ID())
+		// allow replacing EIP155 only when forking (previous rollapp is frozen)
+		if isFound && !rollapp.Frozen {
 			return nil, types.ErrRollappExists
 		}
+		previousRollappChainId, _ := types.NewChainID(rollapp.RollappId)
+		if isFound && rollappId.GetRevisionNumber() != previousRollappChainId.GetRevisionNumber()+1 {
+			return nil, errorsmod.Wrapf(types.ErrInvalidRollappID, "revision number should be %d", previousRollappChainId.GetRevisionNumber()+1)
+		}
```

Comment:
Why not move the `rollappId` checks to a separate method

```suggestion
func (k msgServer) checkIfRollappExists(msg *types.MsgCreateRollapp, ctx sdk.Context) error {
	rollappId, err := types.NewChainID(msg.RollappId)
	if err != nil {
		return err
	}
	// check to see if the RollappId has been registered before
	if _, isFound := k.GetRollapp(ctx, rollappId.GetChainID()); isFound {
		return types.ErrRollappExists
	}

	if !rollappId.IsEIP155() {
		return nil
	}

	// check to see if the RollappId has been registered before with same key
	rollapp, isFound := k.GetRollappByEIP155(ctx, rollappId.GetEIP155ID())
	// allow replacing EIP155 only when forking (previous rollapp is frozen)
	if !isFound {
		return nil
	}

	if !rollapp.Frozen {
		return types.ErrRollappExists
	}

	previousRollappChainId, _ := types.NewChainID(rollapp.RollappId)
	nextRevisionNumber := previousRollappChainId.GetRevisionNumber() + 1

	if rollappId.GetRevisionNumber() != nextRevisionNumber {
		return errorsmod.Wrapf(types.ErrInvalidRollappID, "revision number should be %d", nextRevisionNumber)
	}
	return nil
}

```

Reply:
func added